plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainer.java

```diff
@@ -79,15 +80,15 @@ public String getDescription() {
         }
         entries[idx++] =
             JavaCore.newLibraryEntry(repositoryService.getJarLocation(libraryFile.getMavenCoordinates()),
-                                     repositoryService.getSourceJarLocation(libraryFile.getMavenCoordinates()),
+                                     getSourceLocation(repositoryService, libraryFile),
```

Comment:
I think you're missing the classpath attribute to add the javadoc (if available)